pkg/agents/team3/agentMain.go

```diff
@@ -68,6 +94,19 @@ func (a *CustomAgent3) Run() {
 		default:
 		}
 	}
+
+	//Update variables right after eating
+	if foodTaken != food.FoodType(-1) {
+		a.knowledge.lastHP = a.HP()
+		a.knowledge.foodLastEaten = food.FoodType(foodTaken)
+		a.decisions.foodToEat = -1
+		a.decisions.foodToLeave = -1
+	}
```

Comment:
foodTaken should never be -1 - you should look at the new TakeFood() implementation to see what this should be changed to.